libbeat/common/kubernetes/watcher.go

```diff
@@ -97,16 +97,15 @@ type watcher struct {
 // resource from the cluster (filtered to the given node)
 func NewWatcher(client kubernetes.Interface, resource Resource, opts WatchOptions, indexers cache.Indexers) (Watcher, error) {
 	var store cache.Store
-	var queue workqueue.RateLimitingInterface
+	var queue workqueue.Interface
 
-	informer, objType, err := NewInformer(client, resource, opts, indexers)
+	informer, _, err := NewInformer(client, resource, opts, indexers)
 	if err != nil {
 		return nil, err
 	}
 
 	store = informer.GetStore()
-	queue = workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), objType)
-	ctx, cancel := context.WithCancel(context.Background())
+	queue = workqueue.New()
```

Comment:
Not sure why a rate limited queue is needed here. Removing this for simplification, but happy to move the change to a different PR if preferred.